service.py

```diff
@@ -144,12 +130,14 @@ def get_biospecimen_kf_id(self, external_sample_id, study_id):
         while retry_count > 1:
             resp = requests.get(
                 self.api+'/biospecimens?study_id='+study_id +
-                '&external_sample_id='+external_sample_id)
+                '&external_sample_id='+external_sample_id,
+                timeout=self.context.get_remaining_time_in_millis()-13000)
```

Comment:
Why do these subtracted values get smaller for each request? (14000, 13000, 12000, etc)